IPython/terminal/interactiveshell.py

```diff
@@ -211,11 +227,9 @@ def _displayhook_class_default(self):
     def init_term_title(self, change=None):
         # Enable or disable the terminal title.
         if self.term_title:
-            toggle_set_term_title(True)
             set_term_title(self.term_title_format.format(cwd=abbrev_cwd()))
-        else:
-            toggle_set_term_title(False)
-
+        toggle_set_term_title(self.term_title)
```

Comment:
Seem Ok, here but not strictly equivalent. What if term_title is false-ish ? like empty string,  None, or empty list in a subclass ? Does `toggle_set_term_title` allow non Boolean values ?

Reply:
True, returning into the original state.